File: src/model_v4/entities/SBS_APTYP.ts
import { Column, Entity, Index, OneToMany, PrimaryColumn } from "typeorm";
import { SBS_ADRTYP } from "./SBS_ADRTYP";
import { SBS_APKLASSE } from "./SBS_APKLASSE";
import { SBS_APSTATISTIK } from "./SBS_APSTATISTIK";
import { SBS_HWTYP } from "./SBS_HWTYP";


@Entity("SBS_APTYP",{schema:"sbsdb"})
@Index("APTYP",["APTYP",],{unique:true})
@Index("sbsaptyp_index1",["LFD_NR",])
export class SBS_APTYP {

    @PrimaryColumn("bigint",{ 
        nullable:false,
        primary:true,
        name:"APTYP_INDEX"
        })
    APTYP_INDEX:string;
        

    @Column("varchar",{ 
        nullable:false,
        unique: true,
        length:50,
        name:"APTYP"
        })
    APTYP:string;
        

    @Column("bigint",{ 
        nullable:true,
        default: () => "'NULL'",
        name:"FLAG"
        })
    FLAG:string | null;
        

    @Column("bigint",{ 
        nullable:true,
        default: () => "'NULL'",
        name:"LFD_NR"
        })
    LFD_NR:string | null;
        

    @Column("varchar",{ 
        nullable:true,
        length:200,
        default: () => "'NULL'",
        name:"PARAM"
        })
    PARAM:string | null;
        

   
    @OneToMany(type=>SBS_APKLASSE, SBS_APKLASSE=>SBS_APKLASSE.aPTYP_INDEX,{ onDelete: 'RESTRICT' ,onUpdate: 'RESTRICT' })
    sBS_APKLASSEs:SBS_APKLASSE[];
    

   
    @OneToMany(type=>SBS_APSTATISTIK, SBS_APSTATISTIK=>SBS_APSTATISTIK.aPTYP_INDEX,{ onDelete: 'RESTRICT' ,onUpdate: 'RESTRICT' })
    sBS_APSTATISTIKs:SBS_APSTATISTIK[];
    

   
    @OneToMany(type=>SBS_ADRTYP, SBS_ADRTYP=>SBS_ADRTYP.aPTYP_INDEX,{ onDelete: 'RESTRICT' ,onUpdate: 'RESTRICT' })
    sBS_ADRTYPs:SBS_ADRTYP[];
    

   
    @OneToMany(type=>SBS_HWTYP, SBS_HWTYP=>SBS_HWTYP.aPTYP_INDEX,{ onDelete: 'RESTRICT' ,onUpdate: 'RESTRICT' })
    sBS_HWTYPs:SBS_HWTYP[];
    
}

File: SbsdbServer/sbsdbv4/model/SbsAptyp.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hb.SbsdbServer.sbsdbv4.model {
    [Table("SBS_APTYP")]
    public class SbsAptyp {
        public SbsAptyp() {
            SbsAdrtyp = new HashSet<SbsAdrtyp>();
            SbsApklasse = new HashSet<SbsApklasse>();
            SbsApstatistik = new HashSet<SbsApstatistik>();
            SbsHwtyp = new HashSet<SbsHwtyp>();
        }

        [Column("APTYP_INDEX", TypeName = "bigint(20)")]
        public long AptypIndex { get; set; }

        [Required]
        [Column("APTYP", TypeName = "varchar(50)")]
        public string Aptyp { get; set; }

        [Column("FLAG", TypeName = "bigint(20)")]
        public long? Flag { get; set; }

        [Column("LFD_NR", TypeName = "bigint(20)")]
        public long? LfdNr { get; set; }

        [Column("PARAM", TypeName = "varchar(200)")]
        public string Param { get; set; }

        [InverseProperty("AptypIndexNavigation")]
        public virtual ICollection<SbsAdrtyp> SbsAdrtyp { get; set; }

        [InverseProperty("AptypIndexNavigation")]
        public virtual ICollection<SbsApklasse> SbsApklasse { get; set; }

        [InverseProperty("AptypIndexNavigation")]
        public virtual ICollection<SbsApstatistik> SbsApstatistik { get; set; }

        [InverseProperty("AptypIndexNavigation")]
        public virtual ICollection<SbsHwtyp> SbsHwtyp { get; set; }
    }
}
